doc/architecture/decisions/0002-object-identifiers.md

```diff
@@ -0,0 +1,57 @@
+# 0002. Object identifiers
+
+Date: 2019-06-11
+
+## Status
+
+Agreed
+
+## Context
+
+To uniquely identify an object in Ensembl we need three pieces of information:
+
+-   The **dataset** to which the object belongs, e.g. homo_sapiens_GCA_000001405_14 [^1]
+-   The **object type**, e.g. gene, transcript, region  
+-   The **object ID**, e.g. ENSG00000139618, AT3G52430, 13:32315474-32400266 [^2]  
+  
+Ensembl objects are used extensively in the web client and passing around three separate pieces of information for each one can be cumbersome, error prone, and adds additional work when serialisation is required (e.g. for use in URLs or hash keys).  
+
+[^1]: For our purposes dataset can be considered synonymous to genome_id
+[^2]: Region strings are not IDs but we can consider them that way in this case
+
+## Decision
+
+To allow more convenient manipulation of Ensembl objects we will use a string to uniquely identify each object. These “object IDs” will be generated and parsed by the back-end APIs, and should not be generated or parsed by the client (i.e. the client should not assume anything about the ID format). When the client needs information about an object it will request it through the relevant API calls.
+
+Object ID format
+
+`<dataset>:<object type>:<object id>`
+
+Examples
+```
+homo_sapiens_GCA_000001405_14:gene:ENSG00000139618
+arabidopsis_thaliana_GCA_000001735_1:gene:AT3G52430
+arabidopsis_thaliana_GCA_000001735_1:transcript:AT3G52430
+homo_sapiens_GCA_000001405_14:location:13:32315474-32400266
```

Comment:
Currently, in the frontend code, this object type is called a `region` rather than `location`.